### ISSUE TITLE
How is it different from Spring properties? 

### ISSUE THREAD

**Issue by adrian-scehura** [cdimascio/dotenv-kotlin#39]

**Comment by paulschwarz** [comment]
I would encourage you to read this thread https://github.com/cdimascio/java-dotenv/issues/6 and https://12factor.net/config

Dotenv simply loads variables from the environment, but for developer convenience, it supports loading from your local .env file. This file is not committed to source control, which means that you control your local development environment. For production, you feed in the environment variables via your deployment pipeline.

Spring properties is a sophisticated mechanism to import properties (from wherever) into your java code so you can programmatically utillise those values. Spring supports loading these values from .properties, .yml, but can be extended to load from a remote source, for example.

The project [spring-dotenv](https://github.com/paulschwarz/spring-dotenv) works together with [java-dotenv](https://github.com/cdimascio/java-dotenv) to configure dotenv as a Spring property source.

Does that help?

**Comment by adrian-scehura** [comment]
So the only benefit of using java-dotenv is loading environments from local .env file?

**Comment by paulschwarz** [comment]
TLDR: yes, correct.

"A litmus test for whether an app has all config correctly factored out of the code is whether the codebase could be made open source at any moment, without compromising any credentials." 
  -- https://12factor.net/config

"But it is not always practical to set environment variables on development machines or continuous integration servers where multiple projects are run. dotenv loads variables from a .env file into ENV when the environment is bootstrapped."
  -- https://github.com/bkeepers/dotenv
 
Spring supports configuration profiles, but...
"This method does not scale cleanly: as more deploys of the app are created, new environment names are necessary, such as staging or qa. As the project grows further, developers may add their own special environments like joes-staging, resulting in a combinatorial explosion of config which makes managing deploys of the app very brittle."
  -- https://12factor.net/config

**Comment by adrian-scehura** [comment]
I didn't think about the disadvantages of profiles. Thank you for answers 🎉

**Comment by cdimascio** [comment]
Thanks @paulschwarz
